hail/python/hail/methods/impex.py

```diff
@@ -654,19 +645,19 @@ def import_bed(path, reference_genome='default', skip_invalid_intervals=False) -
 
     if t.row.dtype == tstruct(f0=tstr, f1=tint32, f2=tint32):
         t = t.select(interval=locus_interval_expr(t['f0'],
-                                                  t['f1'] + 1,
-                                                  t['f2'],
-                                                  True,
-                                                  True,
```

Comment:
Because `ReferenceGenome.toLocusInterval` now checks for and fixes the case where start=end, I think this would be fine the way it was, or even better `t['f1']+1, t['f2']+1, True, False`, which I think is most faithful to how the interval was represented in the bed format (just shifting to 1-based indexing). If you think it's okay the way it is now, that's fine with me, but I know you were concerned about changing the representation of intervals to something a user didn't expect.

Reply:
I like this better. Will make the fix.

Reply:
The second coordinate is 1-based and exclusive. The first coordinate is 0-based. So I propose then it should be exactly what we had before. 
https://genome.ucsc.edu/FAQ/FAQformat.html#format1

So it would be `t['f1'] + 1, t['f2'], True, False`

Reply:
I made the change I proposed.

Reply:
If you want to leave it the way it was before, that's `t['f1'] + 1, t['f2'], True, True`.

I defer to you on what is least surprising to bioinformaticists, but adding 1 to one endpoint but not the other just feels wrong. I don't see anything in that BED format FAQ to suggest using 0-based indexing for one endpoint and 1-based for the other. The only discussion I see is

>  chromStart - The starting position of the feature in the chromosome or scaffold. The first base in a chromosome is numbered 0.
chromEnd - The ending position of the feature in the chromosome or scaffold. The chromEnd base is not included in the display of the feature. For example, the first 100 bases of a chromosome are defined as chromStart=0, chromEnd=100, and span the bases numbered 0-99.

which I interpret as [chromStart, chromEnd) in 0-based indexing, or [chromStart+1, chromEnd+1) converting to 1-based indexing.

Anyways, I think you've fixed it so that any of these options are correct, and only affect how the parsed intervals are printed. So I'm okay with whatever you think is best.

Reply:
@konradjk Do you have an opinion on this? Now I'm not certain I'm interpreting the file format correctly.

Reply:
@tpoterba @konradjk Can someone please double check my logic on the interpretation of start and end for BED files? The only thing that is left is making sure we import them correctly from Python.

Reply:
Looking at this [post](http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/), it looks like they suggest adding 1 to the start point to convert to 1-based system. However, if we do that, then we can't handle the case of `1 5 5` as that will cause the interval to be `1:6-5`. So I've left the interval as `(contig:start-end]`. I'm not 100% sure that's what we want.

Reply:
That post is very explicit (more than the docs!) that BED uses a 0-based indexing where intervals are closed on the right and open on the left. So my suggestion to convert to 1-based indexing with half open intervals is consistent, i.e. convert to [start+1, end+1). (start, end] also works.